src/server/journal/tx_executor.cc

```diff
@@ -54,8 +54,10 @@ void TransactionData::AddEntry(journal::ParsedEntry&& entry) {
   opcode = entry.opcode;
 
   switch (entry.opcode) {
-    case journal::Op::PING:
+    case journal::Op::LSN:
```

Comment:
But we said we would extend `PING`. I would have done the same :/

Reply:
I think that extending PING instead of adding another opcode would be the right way. BUT when going into the code I understand that serializer writer can not decide weather to add the lsn data to ping because it is used for all registered callbacks and some of them may support the new format  and some may not. f.e we can have 2 replicas one with ver4 and other with vers3. so serializer which is used in journal silce which first serializes the entry and than call all the register callbacks can not decide if to add the lsn data or not. There for introducing new Opcode is the best solution